File: test/corpus/example/01-rainbow-text-demo/case.ts
import { newOutput, Profile, rgbColor, withProfile } from '../../../../src/index.js';

// Force TrueColor profile for best results
const out = newOutput(process.stdout, withProfile(Profile.TrueColor));

console.log('--- Rainbow Text Demo ---');

// Rainbow colors
const colors = [
  '#FF0000', // Red
  '#FF7F00', // Orange
  '#FFFF00', // Yellow
  '#00FF00', // Green
  '#0000FF', // Blue
  '#4B0082', // Indigo
  '#9400D3', // Violet
];

const text = 'RAINBOW';

// Create rainbow text
let rainbow = '';
for (let i = 0; i < text.length; i++) {
  const char = text[i]!;
  const color = rgbColor(colors[i % colors.length]!);
  const styled = out.string(char).bold().foreground(color);
  rainbow += styled.toString();
}

console.log(`Rainbow Text: ${rainbow}`);

// Rainbow with gradient background
console.log('\n--- Rainbow with Background ---');
let rainbowBg = '';
for (let i = 0; i < text.length; i++) {
  const char = text[i]!;
  const fgColor = rgbColor(colors[i % colors.length]!);
  const bgColor = rgbColor('#000080'); // Navy background
  const styled = out.string(char).bold().foreground(fgColor).background(bgColor);
  rainbowBg += styled.toString();
}

console.log(`Rainbow with BG: ${rainbowBg}`);

// Different styles for each letter
console.log('\n--- Varied Styles ---');
const styles = [
  (s: any) => s.bold(),
  (s: any) => s.italic(),
  (s: any) => s.underline(),
  (s: any) => s.overline(),
  (s: any) => s.reverse(),
  (s: any) => s.blink(),
  (s: any) => s.crossOut(),
];

let variedStyles = '';
for (let i = 0; i < text.length; i++) {
  const char = text[i]!;
  const color = rgbColor(colors[i % colors.length]!);
  let styled = out.string(char).foreground(color);
  styled = styles[i % styles.length]?.(styled);
  variedStyles += styled.toString();
}

console.log(`Varied Styles: ${variedStyles}`);

// Create a border
console.log('\n--- Border Demo ---');
const border = out.string('═══════════════════').bold().foreground(rgbColor('#00FFFF'));
console.log(border.toString());

const centeredText = out
  .string('  TERMENV DEMO  ')
  .bold()
  .foreground(rgbColor('#FFFFFF'))
  .background(rgbColor('#FF0080'));
console.log(centeredText.toString());

console.log(border.toString());

File: test/corpus/example/01-rainbow-text-demo/case.go
package main

import (
	"fmt"

	"github.com/muesli/termenv"
)

func main() {
	// Force TrueColor profile for best results
	out := termenv.NewOutput(nil, termenv.WithProfile(termenv.TrueColor))
	
	fmt.Println("--- Rainbow Text Demo ---")
	
	// Rainbow colors
	colors := []string{
		"#FF0000", // Red
		"#FF7F00", // Orange
		"#FFFF00", // Yellow
		"#00FF00", // Green
		"#0000FF", // Blue
		"#4B0082", // Indigo
		"#9400D3", // Violet
	}
	
	text := "RAINBOW"
	
	// Create rainbow text
	rainbow := ""
	for i, char := range text {
		color := termenv.RGBColor(colors[i%len(colors)])
		styled := out.String(string(char)).Bold().Foreground(color)
		rainbow += fmt.Sprintf("%s", styled)
	}
	
	fmt.Printf("Rainbow Text: %s\n", rainbow)
	
	// Rainbow with gradient background
	fmt.Println("\n--- Rainbow with Background ---")
	rainbowBg := ""
	for i, char := range text {
		fgColor := termenv.RGBColor(colors[i%len(colors)])
		bgColor := termenv.RGBColor("#000080") // Navy background
		styled := out.String(string(char)).Bold().Foreground(fgColor).Background(bgColor)
		rainbowBg += fmt.Sprintf("%s", styled)
	}
	
	fmt.Printf("Rainbow with BG: %s\n", rainbowBg)
	
	// Different styles for each letter
	fmt.Println("\n--- Varied Styles ---")
	styles := []func(termenv.Style) termenv.Style{
		func(s termenv.Style) termenv.Style { return s.Bold() },
		func(s termenv.Style) termenv.Style { return s.Italic() },
		func(s termenv.Style) termenv.Style { return s.Underline() },
		func(s termenv.Style) termenv.Style { return s.Overline() },
		func(s termenv.Style) termenv.Style { return s.Reverse() },
		func(s termenv.Style) termenv.Style { return s.Blink() },
		func(s termenv.Style) termenv.Style { return s.CrossOut() },
	}
	
	variedStyles := ""
	for i, char := range text {
		color := termenv.RGBColor(colors[i%len(colors)])
		styled := out.String(string(char)).Foreground(color)
		styled = styles[i%len(styles)](styled)
		variedStyles += fmt.Sprintf("%s", styled)
	}
	
	fmt.Printf("Varied Styles: %s\n", variedStyles)
	
	// Create a border
	fmt.Println("\n--- Border Demo ---")
	border := out.String("═══════════════════").Bold().Foreground(termenv.RGBColor("#00FFFF"))
	fmt.Printf("%s\n", border)
	
	centeredText := out.String("  TERMENV DEMO  ").Bold().Foreground(termenv.RGBColor("#FFFFFF")).Background(termenv.RGBColor("#FF0080"))
	fmt.Printf("%s\n", centeredText)
	
	fmt.Printf("%s\n", border)
}